File: app/src/main/java/org/tasks/etesync/UpdateEteSyncAccountViewModel.java
package org.tasks.etesync;

import static org.tasks.Strings.isNullOrEmpty;

import androidx.annotation.Nullable;
import androidx.core.util.Pair;
import com.etesync.journalmanager.UserInfoManager.UserInfo;
import org.tasks.ui.CompletableViewModel;

@SuppressWarnings("WeakerAccess")
public class UpdateEteSyncAccountViewModel extends CompletableViewModel<Pair<UserInfo, String>> {
  void updateAccount(
      EteSyncClient client,
      String url,
      String username,
      @Nullable String password,
      @Nullable String token) {
    run(
        () -> {
          client.setForeground();
          if (isNullOrEmpty(password)) {
            return Pair.create(client.forUrl(url, username, null, token).getUserInfo(), token);
          } else {
            String newToken = client.forUrl(url, username, null, null).getToken(password);
            return Pair.create(
                client.forUrl(url, username, null, newToken).getUserInfo(), newToken);
          }
        });
  }
}

File: app/src/main/java/org/tasks/etesync/UpdateEteSyncAccountViewModel.kt
package org.tasks.etesync

import androidx.core.util.Pair
import com.etesync.journalmanager.UserInfoManager.UserInfo
import dagger.hilt.android.lifecycle.HiltViewModel
import org.tasks.Strings.isNullOrEmpty
import org.tasks.ui.CompletableViewModel
import javax.inject.Inject

@Deprecated("use etebase")
@HiltViewModel
class UpdateEteSyncAccountViewModel @Inject constructor(
        private val client: EteSyncClient) : CompletableViewModel<Pair<UserInfo, String>>() {
    suspend fun updateAccount(url: String, user: String, pass: String?, token: String) {
        run {
            client.setForeground()
            if (isNullOrEmpty(pass)) {
                Pair.create(client.forUrl(url, user, null, token).userInfo(), token)
            } else {
                val newToken = client.forUrl(url, user, null, null).getToken(pass)
                Pair.create(client.forUrl(url, user, null, newToken).userInfo(), newToken)
            }
        }
    }
}